lib/grandpa/message_handler.go

```diff
@@ -284,27 +284,32 @@ func (h *MessageHandler) verifyCatchUpResponseCompletability(prevote, precommit
 	return nil
 }
 
-func getEquivocatoryVoters(votes []AuthData) map[ed25519.PublicKeyBytes]struct{} {
+func getEquivocatoryVoters(votes []AuthData) (map[ed25519.PublicKeyBytes]struct{}, error) {
```

Comment:
nit named return values would be nice

```suggestion
func getEquivocatoryVoters(votes []AuthData) (eqvVoters map[ed25519.PublicKeyBytes]struct{}, err error) {
```